unattended_scripts/install_functions/opendistro/common.sh

```diff
@@ -46,48 +46,66 @@ checkSystem() {
 
 checkNames() {
 
-    if [[ "${einame}" == "${kiname}" ]] || [[ "${einame}" == "${winame}" ]] || [[ "${kiname}" == "${winame}" ]]; then
+    if [[ -n ${einame} ]] && [[ -n ${kiname} ]] && ([[ "${einame}" == "${kiname}" ]]); then
         logger -e "The node names for Elastisearch, Kibana and Wazuh must be different."
         exit 1
     fi
 
-    if [ -n ${einame} ]; then
+    if [[ -n ${einame} ]] && [[ -n ${winame} ]] && ([[ "${einame}" == "${winame}" ]]); then
+        logger -e "The node names for Elastisearch, Kibana and Wazuh must be different."
```

Comment:
This logger could be more precise. You know you are comparing `einame` and `winame`

Reply:
Done